es6/jsxcad-ui-threejs.js

```diff
@@ -1375,38 +1382,48 @@ const moveToFit = ({
     control.reset();
   }
 
-  const center = box.getCenter(new Vector3());
-
-  const size = {
-    x: Math.max(Math.abs(box.min.x), Math.abs(box.max.x)),
-    y: Math.max(Math.abs(box.min.y), Math.abs(box.max.y)),
-    z: Math.max(Math.abs(box.min.z), Math.abs(box.max.z)),
-  };
-
-  const maxSize = Math.max(size.x, size.y, size.z);
-  const fitHeightDistance =
-    maxSize / (2 * Math.atan((Math.PI * camera.fov) / 360));
-  const fitWidthDistance = fitHeightDistance / (camera.aspect || 1);
-  const zoomOut = 1.5;
-  const distance =
-    fitOffset * Math.max(fitHeightDistance, fitWidthDistance) * zoomOut;
-
   const target = new Vector3(0, 0, 0);
 
-  const direction = target
-    .clone()
-    .sub(camera.position)
-    .normalize()
-    .multiplyScalar(distance);
-
-  camera.near = distance / 100;
-  camera.far = distance * 100;
-  camera.updateProjectionMatrix();
+  const points = [
+    new Vector3(box.min.x, box.min.y, box.min.z),
+    new Vector3(box.max.x, box.min.y, box.min.z),
+    new Vector3(box.min.x, box.max.y, box.min.z),
+    new Vector3(box.max.x, box.max.y, box.min.z),
+    new Vector3(box.min.x, box.min.y, box.max.z),
+    new Vector3(box.max.x, box.min.y, box.max.z),
+    new Vector3(box.min.x, box.max.y, box.max.z),
+    new Vector3(box.max.x, box.max.y, box.max.z),
+  ];
 
-  camera.position.copy(center).sub(direction);
+  // Back the camera off by 1mm at a time until the bounding box is contained by
+  // the frustrum.
+  // CHECK: Use a binary search if this becomes a latency problem.
+  for (let distance = 1; ; distance += 1) {
+    const direction = target
+      .clone()
+      .sub(camera.position)
+      .normalize()
+      .multiplyScalar(distance);
+
+    camera.position.copy(target).sub(direction);
+
+    camera.near = distance / 100;
```

Comment:
I was seeing an issue where if I created an orbitView() with a small object in it and then updated it with a larger object the near and far planes would be wrong so part of the object could get cut off. Does this run just once when the orbitView is created, or every time it is updated?

Reply:
Just when it's created -- the camera should be persistent through updates, so that you can see changes in-place.

Reply:
Could we have a way to update just the near and far distances? I'm finding situations where I create a view of something small and then it grows as I work on it. Eventually parts of the geometry stick out past the far plane and aren't rendered anymore

Reply:
That would probably do it. I have made things bigger than 10m (house model).

Reply:
Let's try 1 km, and see how that goes. :)